packages/backend/src/modules/metric/metric.service.ts

```diff
@@ -0,0 +1,168 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { config } from 'node-config-ts';
+import { Redis } from 'ioredis';
+import { MetricsChannels } from '@graphql-portal/types';
+import { LoggerService } from '../../common/logger';
+import { IRequestMetricDocument } from '../../data/schema/request-metric.schema';
+import { INetworkMetricDocument } from '../../data/schema/network-metric.schema';
+import Provider from '../../common/enum/provider.enum';
+import {
+  AnyMetric,
+  AnyResolverMetric,
+  IGotError,
+  IGotRequest,
+  ISentResponse,
+  IReducedResolver,
+} from './interfaces';
+
+@Injectable()
+export default class MetricService {
+  private redis: Redis;
+  private intervals: NodeJS.Timer[] = [];
+
+  public constructor(
+    @Inject(Provider.REDIS) private readonly redisClients: [Redis, Redis],
+    @InjectModel('RequestMetric') private requestMetricModel: Model<IRequestMetricDocument>,
+    @InjectModel('NetworkMetric') private networkMetricModel: Model<INetworkMetricDocument>,
+    private readonly logger: LoggerService,
+  ) {
+    [this.redis] = this.redisClients;
+  }
+
+  private onModuleInit(): void {
+    this.init();
```

Comment:
why init body is not here?

Reply:
it's more clear to have dedicated func, onModuleInit can have many calls inside